pulsar-client-go/pulsar/c_producer.go

```diff
@@ -156,7 +156,7 @@ func createProducerAsync(client *client, schema Schema, options ProducerOptions,
 		C._pulsar_producer_configuration_set_message_router(conf, savePointer(&options.MessageRouter))
 	}
 
-	if options.Batching {
+	if !options.Batching {
```

Comment:
This is a different issue which we should address separately. The main issue is that batching is just a flag, so there's no way to know wether the user touched it or not (eg: it should have been a `*bool` instead).

Reply:
In cpp client, the default value of batch is true. So, only when batch is false, the user need to set it up. 

> (eg: it should have been a `*bool` instead)

yes, use this way, we can know wether the user touched it or not, by: 

```
if options.Batching !=nil {
    ...
}
```

but the effect between them is the same. 

When using it, I think users prefer to use it this way:

```
producer, err := client.CreateProducer(ProducerOptions{
	Topic:    topic,
	Batching: false,
})
```

Instead of:

```
var flag  = false
producer, err := client.CreateProducer(ProducerOptions{
	Topic:    topic,
	Batching: &flag,
})
```

So, I don't think we need to care wether the user has `batch` here.